readthedocs/api/v3/filters.py

```diff
@@ -7,9 +7,14 @@
 
 class ProjectFilter(filters.FilterSet):
 
+    slug = filters.CharFilter(lookup_expr='icontains')
+    name = filters.CharFilter(lookup_expr='icontains')
```

Comment:
Do we want `contains` or `startswith`?

Reply:
The user who asked for this wanted contains, I think that makes sense mostly when you have long names: "Test", "Project test", "Testing". So "test" would match all three.

Reply:
Contains will help more as we have no Idea about the starting text most of the times. As My document name could be "how to check card details" where Customer can come and search with name "card details" so contains will work as expected:)

Reply:
If we can consider different words too , that will be best. Like - If my document name is "How to handle my payment card securely" , user comes and searches "handle payment securely" . If we are doing full text contains search this will not give me the above document. But if we can split the string coming from customer and then can search like- Customer query is "handle payment securely" we break it into tokens ["handle", "payment", "securely"] and we can get all the docs which have all these words in their names.

Reply:
@chadwhitacre we use ES for that kind of searches https://readthedocs.org/search/?q=docs&type=project

We don't expose the API yet, but we will in the future

Reply:
Hey all. 👋  &nbsp; Hope you're holding up okay in These Unprecedented Times&trade;.

Bump @chdharm.

Reply:
@stsewd Thank you for much for letting me know but at-least this "contains" based implementation is going be merged right? @chadwhitacre Yup. How you all are safe too. Thank you so much everyone. I would love to contribute into this project.

Reply:
@chdharm the filter proposed in this PR is to filter "project names" or "project slugs". From your examples, it seems those are examples of "page names" where this filter won't have any effect. Is my interpretation correct?

Reply:
Yup, 100%. @humitos